### PR TITLE
Upgraded plugin for Strapi 4.20 and NodeJS 20

### LINKED ISSUE #37: Could not load js config file D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\strapi-server.js: Cannot find module '@strapi/utils/dist/file'
Error: Could not load js config file D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\strapi-server.js: Cannot find module '@strapi/utils/dist/file'   │    
│   Require stack:                                                                                                                                                                 │    
│   - D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\dist\server\services\image-optimizer-service.js                                                     │    
│   - D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\dist\server\services\index.js                                                                       │    
│   - D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\dist\server\index.js                                                                                │    
│   - D:\projects\strapi\my-project\node_modules\strapi-plugin-image-optimizer\strapi-server.js                                                                                    │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\utils\dist\index.js                                                                                                       │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\commands\actions\admin\create-user\action.js                                                                  │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\commands\actions\admin\create-user\command.js                                                                 │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\commands\index.js                                                                                             │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\cli.js                                                                                                        │    
│   - D:\projects\strapi\my-project\node_modules\@strapi\strapi\bin\strapi.js                                                                                                      │    
│       at loadJsFile (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\core\app-configuration\load-config-file.js:18:13)                                            │    
│       at Module.loadFile (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\core\app-configuration\load-config-file.js:37:14)                                       │    
│       at Object.loadPlugins (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\core\loaders\plugins\index.js:90:41)                                                 │    
│       at async Strapi.loadPlugins (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\Strapi.js:310:5)                                                               │    
│       at async Promise.all (index 3)                                                                                                                                             │    
│       at async Strapi.register (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\Strapi.js:340:5)                                                                  │    
│       at async Strapi.load (D:\projects\strapi\my-project\node_modules\@strapi\strapi\dist\Strapi.js:424:5)                                                                      │    
│       at async Object.develop (D:\projects\strapi\my-project\node_modules\@strapi\admin\dist\_chunks\index-ffd2f664.js:1245:28)                                                  │    
│       at async develop (D:\projects\strapi\my-project\node_modules\@strapi\admin\dist\_chunks\develop-da585b1e.js:52:5)                                                          │    
│       at async Command.parseAsync (D:\projects\strapi\my-project\node_modules\commander\lib\command.js:923:5) 

**I can't use the plugin, can you tell me what it is?**

image

image

image

image

### PR DESCRIPTION
**Pull request by BirknerAlex**
Fixes #37

### CODE REVIEW

**Review by yarikwest — APPROVED** [review]

**Review by yarikwest — APPROVED** [review]
@BirknerAlex  looks good to me!
@marlokessler can we merge this changes?

**Comment by yarikwest** [comment]
@marlokessler can we merge this changes?

**Review by marlokessler — APPROVED** [review]
Hi @BirknerAlex thank you for your contribution! Looks good to me!

**Comment by marlokessler** [comment]
@all-contributors please add @BirknerAlex for bug and code.

File: tsconfig.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "compilerOptions": {
     "outDir": "dist",
     "rootDir": ".",
-    "declaration": true
+    "declaration": true,
```
Comment (welschmoor) [inline tsconfig.json:6]: trailing comma in json not good

Comment (marlokessler) [reply tsconfig.json:6]: Oh true, good you catched that!

File: server/services/image-optimizer-service.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import { ReadStream, createReadStream, createWriteStream } from "fs";
 import { join } from "path";
 import sharp, { Sharp, Metadata } from "sharp";
+import { file as fileUtils } from '@strapi/utils';
```
[the remark this answers is not in the document]
Comment (BirknerAlex) [reply server/services/image-optimizer-service.ts:4]: I just checked the latest source code where the utilty function has been moved. 

https://github.com/search?q=repo%3Astrapi%2Fstrapi%20bytesToKbytes&type=code

**Pull request merged**